recipes/msgpack-c/meta.yaml

```diff
@@ -0,0 +1,69 @@
+{% set name = "msgpack-c" %}
+{% set version = "2.0.0" %}
+{% set sha256 = "eb20b4bf15f20bad149ec82fffac74f16de2a8a797e321a3f8189d63263a511e" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: cpp-{{ version }}.tar.gz
+  url: https://github.com/msgpack/{{ name }}/archive/cpp-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # Support latest gtest
+    # https://github.com/msgpack/msgpack-c/commit/4a1db60af8d6bdb258a34352f5cd8d9800544372.patch
+    - gtest.patch
+    # This is needed for a bug in gcc version used where isnan is ambiguous. (::isnan and std::isnan)
+    - ambiguous_nan_inf.patch   # [linux]
```

Comment:
I think msgpack-c is including `math.h` because it is testing the C interface in `test/msgpack_c.cpp`. They do it correctly on `test/msgpack_basic.cpp` which tests the C++ interface. I know that including `cmath` will work, but I didn't do it because I thought the difference between `test/msgpack_basic.cpp`  and `test/msgpack_c.cpp`  was intentional, which might not be true.